### ISSUE TITLE
MacOS 10.13.1 Configure Error

### ISSUE THREAD

**Issue by alphafan** [meta-toolkit/meta#191]
I have an error when executing command 
```
CXX=clang++ cmake ../ -DCMAKE_BUILD_TYPE=Release -DICU_ROOT=/usr/local/opt/icu4c
```

The error message is below.
I have a mac of 10.13.1. I have tried the develop branch and clean the cache, but it still not working. Could you please help me debug this problem? Thanks.

```
-- The C compiler identification is AppleClang 9.0.0.9000038
-- The CXX compiler identification is AppleClang 9.0.0.9000038
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11") 
-- Looking for lzma_auto_decoder in /usr/lib/liblzma.dylib
-- Looking for lzma_auto_decoder in /usr/lib/liblzma.dylib - found
-- Looking for lzma_easy_encoder in /usr/lib/liblzma.dylib
-- Looking for lzma_easy_encoder in /usr/lib/liblzma.dylib - found
-- Looking for lzma_lzma_preset in /usr/lib/liblzma.dylib
-- Looking for lzma_lzma_preset in /usr/lib/liblzma.dylib - found
-- Could NOT find LibLZMA (missing: LIBLZMA_INCLUDE_DIR) 
-- Searching for ICU 58.2
-- Found ICU: /usr/local/opt/icu4c/lib/libicudata.dylib;/usr/local/opt/icu4c/lib/libicui18n.dylib;/usr/local/opt/icu4c/lib/libicuuc.dylib (Required is at least version "58.2") 
-- ICU version found is 59.1.0, expected 58.2; attempting to build ICU from scratch...
-- ICU include dirs: /Users/yufan/Documents/workspace/meta/build/deps/icu-58.2/include
-- ICU libraries: icui18n;icuuc;icudata
-- Locating libc++...
-- Located libc++: /usr/lib/libc++.dylib
-- Located libc++ include path: /Library/Developer/CommandLineTools/usr/include/c++/v1/
--     Locating libc++'s abi...
--     Found libc++abi: /usr/lib/libc++abi.dylib
-- Found LIBCXX: /usr/lib/libc++.dylib  
-- Using regular malloc; consider installing jemalloc
-- Performing Test META_HAS_ALIGNED_ALLOC
-- Performing Test META_HAS_ALIGNED_ALLOC - Failed
-- Performing Test META_HAS_POSIX_MEMALIGN
-- Performing Test META_HAS_POSIX_MEMALIGN - Failed
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)


-- Configuring incomplete, errors occurred!
See also "/Users/yufan/Documents/workspace/meta/build/CMakeFiles/CMakeOutput.log".
See also "/Users/yufan/Documents/workspace/meta/build/CMakeFiles/CMakeError.log".
```

**Comment by alphafan** [comment]
It seems to me it's an ICU version problem. Do you have any idea how to resolve this problem? Thanks.

**Comment by Yevgnen** [comment]
@alphafan Have you solved this?

**Comment by alphafan** [comment]
@Yevgnen No. Not resolved.

**Comment by slevental** [comment]
The problem is with include path generated by `FindLIBCXX.cmake`, it doesn't find correct installation directory which is shown in `clang++ -v`, instead it finds `/Library/Developer/CommandLineTools/usr/include` and uses it in includes while compiling aligned allocation test code (and other code eventually), you can add original installation path to your clang by adding it as a prefix (not sure why libc++ location /Library/Developer/CommandLineTools/usr/include is not working):
```
cmake -DCMAKE_SYSTEM_PREFIX_PATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include;${CMAKE_SYSTEM_PREFIX_PATH} .. && make -j8
```

**Comment by eklavyamirani** [comment]
on MacOS 10.12, I faced the same issue. My CMAKE_SYSTEM_PREFIX_PATH points to /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include correctly, i.e. 
`debug: CMAKE_SYSTEM_PREFIX_PATH = /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include;/usr/local;/usr;/;/usr/local/Cellar/cmake/3.10.0;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local`


`debug: LIBCXX_LIB_PATH = /usr/lib`

but that didn't work. So I went in to check what was going in AlignedAlloc kludges and noticed it was not able to compile `::posix_memalign(&ptr, 64, 128);` while my standard AppleClang (v9) could compile and execute this code. So I used AppleClang compiler and bypassed the ENABLE_LIBCXX by setting it to OFF in CMakeLists.txt and built again. VOILA! Successful build and all unit tests passed.

**TLDR: if you are getting this error on your mac, modify [ENABLE_LIBCXX to be off](https://github.com/meta-toolkit/meta/blob/master/CMakeLists.txt#L4) and that should build it.**

@alphafan Can you check if this works for you?

**Comment by alphafan** [comment]
@eklavyamirani Thanks very much. After I change the line that you mentioned. (ON -> OFF), it works for me now.

**Comment by YenKang** [comment]
@alphafan I met the similar problem with u, and I have already set **ENABLE_LIBCXX** from **ON** to **OFF**. It still did not work.

![mouse_locator_ _cmakelists_txt_ _meta_ _ _ _2__yen_yende-macbook-pro____meta_build__zsh_](https://user-images.githubusercontent.com/16321107/41329731-f170050c-6f01-11e8-87e6-3dd97e7d0ae1.jpg)

![2018-06-13 11 42 34](https://user-images.githubusercontent.com/16321107/41329740-fa374e52-6f01-11e8-813d-3061bc075056.png)

**Comment by skystrife** [comment]
Fixed in 92fd041.
